### PR TITLE
Merge Thornado coupling into development

### LINKED ISSUE #429: Improve species and aux data for Castro/Thornado
Regarding the ecp/two-moment branch...

For Thornado, we only need to track the electron number density N_e.

Right now N_e is an auxiliary variable and for EOS calls we're getting it from y_e, which we say is the first species. This is a workaround for the fact that we do not load auxiliary variables into `eos_t % aux` before EOS calls in Castro, so it is related to #144 .

To fix this, we should:

1) Enable nspec=0
2) Take out y_e, just put N_e (electron number density) in aux
3) load aux into eos_state before EOS calls
4) in the EOS or `eos_override`, convert N_e to y_e

We can also consider tracking y_e in Castro and converting to/from N_e when calling Thornado.

### PR DESCRIPTION
**Pull request by dwillcox**
This PR merges the Castro/Thornado coupling code into the development branch, now that Thornado, Weaklib, and the EOS/Opacity tables are publicly available.

Thornado public repository (BSD-3): https://github.com/endeve/thornado

Weaklib public repository (BSD-3): https://github.com/starkiller-astro/weaklib

EOS/Opacity tables repository (license unknown): https://code.ornl.gov/m2o/thornado-tables

This PR is not yet ready to merge, but for now it's a good place to brainstorm things we need to do.

Off the top of my head:

- [ ] Add a summary of the Castro/Thornado coupling to the docs #684 
- [ ] Add usage notes to the docs for the test problems #685 
- [ ] Add regression tests for Castro/Thornado test problems #686
- [ ] Merge the `ecp-two-moment` branch in Microphysics into `development` (https://github.com/starkiller-astro/Microphysics/pull/197)
- [ ] Fix #429 to improve handling of aux data
- more?

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite
- [ ] all functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Exec/Make.Castro [hunk 1]
```diff
@@ -373,12 +448,24 @@ ifeq ($(USE_HYPRE), TRUE)
 endif
 
 ifeq ($(USE_HDF5), TRUE)
-  INCLUDE_LOCATIONS += $(HDF5_DIR)/include
-  INCLUDE_LOCATIONS += $(HDF5_INCL)
-  LIBRARY_LOCATIONS += $(HDF5_DIR)/lib
-  LIBRARIES += -lhdf5 -lhdf5_fortran -lhdf5 -lz
+  DEFINES += -DBL_HDF5
+  INCLUDE_LOCATIONS += $(HDF5_HOME)/include 
+  ifneq ($(HOST), manda)
```
Comment (zingale) [inline Exec/Make.Castro]: I don't think we want specific machines hardcoded into the general make file

File: Exec/reacting_tests/bubble_convergence/GNUmakefile [hunk 2]
```diff
@@ -7,7 +7,7 @@ COMP	         = gnu
 
 USE_MPI          = TRUE
 USE_GRAV         = TRUE
-USE_REACT        = TRUE
+USE_REACT        = FALSE
```
Comment (zingale) [inline Exec/reacting_tests/bubble_convergence/GNUmakefile]: why was this changed?

Comment (dwillcox) [reply Exec/reacting_tests/bubble_convergence/GNUmakefile]: That was also changed in a9e7943d74, probably by accident since it was a large commit.

Pushing a change to revert this now ...

File: Exec/Make.Castro (hunk 1 above)
Comment (dwillcox) [reply Exec/Make.Castro]: This section has two machines hardcoded: `manda` and someone's Mac judging by the `apple-darwin` in the path.

I will remove these. Presumably this can be done by manually adding to `INCLUDE_LOCATIONS` and `LIBRARIES` on the `make` line.

Comment (dwillcox) [reply Exec/Make.Castro]: Done in 97ad4d5078f18da302b2407f9eb8b7bb14d32081

File: Exec/reacting_tests/bubble_convergence/GNUmakefile (hunk 2 above)
Comment (dwillcox) [reply Exec/reacting_tests/bubble_convergence/GNUmakefile]: Done in db1be2b6a0af99a96433c59fec178a2b85ba2c64

**Comment by maxpkatz** [comment]
What should we do with the coupling between these codes? Does Thornado become a submodule of Castro?

**Comment by zingale** [comment]
closing since we've diverged so much since this PR that it is probably easiest to start over.
